packages/manager/test/__testutils__/expectHookHandlerToHaveBeenCalledWithData.ts

```diff
@@ -0,0 +1,18 @@
+import { expect, Mock } from "vitest";
+
+// TODO: Replace this function with a custom matcher
+// (e.g. `expect(hookHandler).toHaveBeenCalledWithHookData({ ... })`).
```

Comment:
It would, but that feels messier than using the dedicated `toHaveBeenCalledWith()` matcher.

Using `calls[0][0]` accesses the internal structure of spies, which potentially could change in the future. Using `toHaveBeenCalledWith()` is a stable API that is unaware of the underlying implementation.